### PR TITLE
Implemented new AddMultiTargetCompatibleSampleDocs target

### LINKED ISSUE #5: [Sample App] Non-WASM supported components showing up in WASM Sample App
### Describe the bug

Components like `Shimmer` which use composition and aren't supported in WASM, have a Multi-target that defines as such. However, it's still appearing in the WASM sample app when generating the all solution.

This is tangentially related to the local component head for WASM being copied as well when it's not supported (not sure if we have another issue for that, maybe want a separate one?).

### Steps to reproduce

```text
1. GenerateAllSolution
2. Run WASM head
3. Open controls, see Shimmer
4. Click on Shimmer
5. See error in output as Shimmer page doesn't exist for WASM, shouldn't be listed in app.
```

### Expected behavior

Shimmer not shown in WASM app head.

### Screenshots

_No response_

### Code Platform

- [ ] UWP
- [ ] WinAppSDK / WinUI 3
- [X] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

_No response_

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

_No response_

### Visual Studio Build Number

_No response_

### Device form factor

_No response_

### Additional context

_No response_

### Help us help you

Yes, but only if others can assist.

### PR DESCRIPTION
**Pull request by Arlodotexe**
This PR:
- Fixes #5.
- Adds a new MSBuild inline task `CheckMultiTarget` to verify multi-target compatibility for the files provided to it.
- Introduces a new target `AddMultiTargetCompatibleSampleDocs` which filters markdown files based on platform compatibility.
- Includes error handling and logging in the MSBuild tasks to provide clearer feedback during build processes.
- Excludes markdown files from `obj` and `bin` directories to prevent processing unnecessary files.
- Has been tested on UWP and WASM.

### CODE REVIEW

**Comment by michael-hawker** [comment]
I realized I haven't opened a PR for the doc export script: https://github.com/CommunityToolkit/Tooling-Windows-Submodule/blob/llama/export-docs/Scripts/ExportDocs.ps1

One thing we'll have to think about is how we mark which platforms are supported within the docs and that script. Like for now for WASM it's easiest to remove the markdown files, but we may want to modify this approach to inject the supported platforms into the doc and possible just turn-off sample compilation for unsupported platforms or something.

What that be something we can add in the future with this approach or require other tooling?

File: ProjectHeads/App.Head.props
Comment (michael-hawker) [inline ProjectHeads/App.Head.props:1]: Line 152 has the single sample head markdown file inclusion, so we need to worry about that one too, eh? We may need to think about doing more logic in the solution generation powershell script to exclude the WASM head on components that don't support WASM as well?

File: ProjectHeads/App.Head.props [hunk 1]
```diff
@@ -76,6 +76,48 @@
       </ItemGroup>
     </When>
   </Choose>
+
+  <!-- Gather sample markdown for components that MultiTarget this platform. -->
+  <!-- Each deployable sample app head uses the CommunityToolkit.Toolkit.SampleGen source generator to generate document registries -->
+  <Import Project="$(ToolingDirectory)\ProjectHeads\Tasks\CheckMultiTarget.props" />
+
+  <!-- Simplifies comparison of the current head against component-level <MultiTarget> definition -->
```
Comment (michael-hawker) [inline ProjectHeads/App.Head.props:84]: This file is getting pretty big now, should we either move this into its own props file as well (just from an organizatinal perspective)? (I do like just having the task itself in a single file.)

File: ProjectHeads/App.Head.props
Comment (Arlodotexe) [reply ProjectHeads/App.Head.props:1]: The single component solution should work fine. If a component doesn't support WASM, the head shouldn't even be showing (related #17).

**Comment by Arlodotexe** [comment]
> One thing we'll have to think about is how we mark which platforms are supported within the docs and that script. Like for now for WASM it's easiest to remove the markdown files, but we may want to modify this approach to inject the supported platforms into the doc and possible just turn-off sample compilation for unsupported platforms or something.

The script you've linked is for modifying the documents before exporting for publishing on learn.microsoft.com. It replaces our custom `![SAMPLE_NAME]` syntax with the content of the sample and updates all learn.microsoft.com links to be relative.

It sounds like you want to display which MultiTarget or TargetFramework a component is supported on from the learn.microsoft.com side of things for our users. Would this be as inline text, as metadata, or something else? This would be in addition to the information available on the "Frameworks" tab shown on nuget.org.

We'll need to also display this information in the header for the sample gallery. For display purposes, we should avoid doing this inline. That means we'll need the following:
 
1. The DocumentationRenderer needs to identify what `MultiTargets` are supported for the component a given document is contained in.
2.  The doc export script (the one linked, not yet in the repo) provided will need to pull from `MultiTarget.props` and inject it as front-matter, which can be consumed and rendered by docfx. We can reuse existing code in `GenerateMultiTargetAwareProjectReferenceProps.ps1` for this, but it's not yet clear how we'd display this on the learn.microsoft.com side of things.

Once we have these, we can display the supported MultiTarget or TargetFrameworks for a component within both the sample gallery and the published documentation.

**Review by michael-hawker — APPROVED** [review]
Seems to work well. Later we'll want to adjust the template/solution generation steps to not include the single-WASM head in the components that won't support it. Same for things like UWP in the future if we have something that is only built against WASDK.

**Pull request merged**